components/config/src/config/markup.rs

```diff
@@ -43,26 +49,92 @@ pub struct Markdown {
     pub external_links_no_referrer: bool,
     /// Whether smart punctuation is enabled (changing quotes, dashes, dots etc in their typographic form)
     pub smart_punctuation: bool,
-
-    /// A list of directories to search for additional `.sublime-syntax` files in.
-    pub extra_syntaxes: Vec<String>,
+    /// A list of directories to search for additional `.sublime-syntax` and `.tmTheme` files in.
+    pub extra_syntaxes_and_themes: Vec<String>,
     /// The compiled extra syntaxes into a syntax set
     #[serde(skip_serializing, skip_deserializing)] // not a typo, 2 are need
     pub extra_syntax_set: Option<SyntaxSet>,
+    /// The compiled extra themes into a theme set
+    #[serde(skip_serializing, skip_deserializing)] // not a typo, 2 are need
+    pub extra_theme_set: Arc<Option<ThemeSet>>,
 }
 
 impl Markdown {
-    /// Attempt to load any extra syntax found in the extra syntaxes of the config
-    pub fn load_extra_syntaxes(&mut self, base_path: &Path) -> Result<()> {
-        if self.extra_syntaxes.is_empty() {
-            return Ok(());
+    /// Gets the configured highlight theme from the THEME_SET or the config's extra_theme_set
+    /// Returns None if the configured highlighting theme is set to use css
+    pub fn get_highlight_theme(&self) -> Option<&Theme> {
+        if self.highlight_theme == "css" {
+            None
+        } else {
+            Some(self.get_highlight_theme_by_name(&self.highlight_theme))
+        }
+    }
+
+    /// Gets an arbitrary theme from the THEME_SET or the extra_theme_set
+    pub fn get_highlight_theme_by_name<'config>(&'config self, theme_name: &str) -> &'config Theme {
+        (*self.extra_theme_set)
+            .as_ref()
+            .and_then(|ts| ts.themes.get(theme_name))
+            .unwrap_or_else(|| &THEME_SET.themes[theme_name])
+    }
+
+    /// Attempt to load any extra syntaxes and themes found in the extra_syntaxes_and_themes folders
+    pub fn load_extra_syntaxes_and_highlight_themes(
+        &self,
+        base_path: &Path,
+    ) -> Result<(Option<SyntaxSet>, Option<ThemeSet>)> {
+        if self.extra_syntaxes_and_themes.is_empty() {
+            return Ok((None, None));
         }
 
         let mut ss = SyntaxSetBuilder::new();
-        for dir in &self.extra_syntaxes {
+        let mut ts = ThemeSet::new();
+        for dir in &self.extra_syntaxes_and_themes {
             ss.add_from_folder(base_path.join(dir), true)?;
+            ts.add_from_folder(base_path.join(dir))?;
+        }
+        let ss = ss.build();
+
+        Ok((
+            if ss.syntaxes().is_empty() { None } else { Some(ss) },
+            if ts.themes.is_empty() { None } else { Some(ts) },
+        ))
+    }
+
+    pub fn export_theme_css(&self, theme_name: &str) -> String {
```

Comment:
I would prefer if that stayed with the rest of the highlighting code so syntect doesn't bleed too much in the config but it doesn't seem doable anymore :(